R/derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,528 @@
+#'   Otherwise, no `"SD"` assessment must occur between the two assessments.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#' @param missing_as_ne Consider no assessments as `"NE"`?
+#'
+#'   If the argument is set to `TRUE`, the response is set to `"NE"` for
+#'   subjects without an assessment in the input dataset. Otherwise, the
+#'   response is set to `"MISSING"` for these subjects.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#' @param aval_fun Function to map character analysis value (`AVALC`) to numeric
+#'   analysis value (`AVAL`)
+#'
+#'   The (first) argument of the function must expect a character vector and the
+#'   function must return a numeric vector.
+#'
+#'   *Default:* `aval_resp` (see `aval_resp()` for details)
+#'
+#' @param set_values_to Variables to set
+#'
+#'   A named list returned by `vars()` defining the variables to be set for the
+#'   new parameter, e.g. `vars(PARAMCD = "CBOR", PARAM = "Confirmed Best Overall
+#'   Response")` is expected. The values must be symbols, character strings,
+#'   numeric values, or `NA`.
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   A list of symbols created using `vars()` is expected.
+#'
+#' @details
+#'
+#'   1. The input dataset (`dataset`) is restricted to the observations matching
+#'   `filter_source` and to observations before or at the date specified by
+#'   `source_pd`.
+#'
+#'   1. The following potential confirmed responses are select from the restricted input dataset:
+#'
+#'       - `"CR"`: An assessment is considered as complete response (CR) if
+#'           - `AVALC == "CR"`,
+#'           - there is a confirmatory assessment with `AVALC == "CR"` at least
+#'           `ref_confirm` days after the assessment,
+#'           - all assessments between the assessment and the confirmatory
+#'           assessment are `"CR"` or `"NE"`, and
+#'           - there are at most `max_nr_ne` `"NE"` assessments between the
+#'           assessment and the confirmatory assessment.
+#'
+#'       - `"PR"`: An assessment is considered as partial response (PR) if
+#'           - `AVALC == "PR"`,
+#'           - there is a confirmatory assessment with `AVALC %in% c("CR",
+#'           "PR")` at least `ref_confirm` days after the assessment,
+#'           - all assessments between the assessment and the confirmatory
+#'           assessment are `"CR"`, `"PR"`, or `"NE"`,
+#'           - there is no `"PR"` assessment after a `"CR"` assessment in the
+#'           confirmation period, and
+#'           - there are at most `max_nr_ne` `"NE"` assessments between the
+#'           assessment and the confirmatory assessment.
+#'
+#'           If the `accept_sd` argument is set to `TRUE`, one `"SD"` assessment
+#'           in the confirmation period is accepted.
+#'
+#'        - `"SD"`: An assessment is considered as stable disease (SD) if
+#'          - `AVALC %in% c("CR", "PR", "SD")` and
```

Comment:
should we add here the following?
- no confirmatory `"CR"` or `"PR"` assessment exists

Reply:
I wouldn't add it because it wouldn't match the code and it would make the description longer.

Reply:
ok, i guess it is covered by the fact that later on you explain the hierarchy and if there was confirmed CR or PR then it would already get chosen as best anyway